eva/experimental/ray/executor/ray_stage.py

```diff
@@ -15,28 +15,35 @@
 from typing import Callable, List
 
 import ray
+from ray.exceptions import RayError
 from ray.util.queue import Queue
 
+from eva.executor.executor_utils import ExecutorError
+
 
 class StageCompleteSignal:
     pass
 
 
 @ray.remote(num_cpus=0)
 def ray_stage_wait_and_alert(tasks: ray.ObjectRef, output_queue: Queue):
-    ray.get(tasks)
-    for q in output_queue:
-        q.put(StageCompleteSignal)
+    try:
+        ray.get(tasks)
+        for q in output_queue:
+            q.put(StageCompleteSignal)
+    except RayError as e:
+        for q in output_queue:
+            q.put(ExecutorError(e.cause))
```

Comment:
Consider the following set up
```mermaid
graph TD;
      A[Projection with parallelism=1]--> B[UDF with parallelism = 2];
      B --> C[Get with parallelism=1];
```
, and there is ExecutorError raised at the bottom Get operator. 
If I understand correctly, what will happen is that one ExecutorError will be put into the queue between UDF and Get. And one Actor will get the ExecutorError and due to the fact that it is not root, the [code](https://github.com/georgia-tech-db/eva/blob/9aa9fac520a264dfc3e26583a59d816b8bf53c8d/eva/experimental/ray/executor/exchange_executor.py#L97) will not come into play. So it will try to process on the ExecutorError object, which is not expected. Furthermore, the second actor will not be affected, because it does not get the ExecutorError, which has already been read by the first actor. I think you want to modify the [QueueReaderExecutor](https://github.com/georgia-tech-db/eva/blob/9aa9fac520a264dfc3e26583a59d816b8bf53c8d/eva/experimental/ray/executor/exchange_executor.py#L44) instead.